lms/static/scripts/frontend_apps/index.js

```diff
@@ -6,21 +6,33 @@ import { createElement, render } from 'preact';
 
 import { Config } from './config';
 import BasicLtiLaunchApp from './components/BasicLtiLaunchApp';
+import CanvasAuthErrorDialog from './components/CanvasAuthErrorDialog';
 import FilePickerApp from './components/FilePickerApp';
 import { startRpcServer } from '../postmessage_json_rpc/server';
 
-// Create an RPC Server and start listening to postMessage calls.
-const rpcServer = startRpcServer();
-
 const rootEl = document.querySelector('#app');
 const config = JSON.parse(document.querySelector('.js-config').textContent);
 
+let rpcServer;
+if (config.mode === 'basic-lti-launch') {
+  // Create an RPC Server and start listening to postMessage calls.
+  rpcServer = startRpcServer();
+}
+
 render(
   <Config.Provider value={config}>
     {config.mode === 'basic-lti-launch' && (
       <BasicLtiLaunchApp rpcServer={rpcServer} />
     )}
     {config.mode === 'content-item-selection' && <FilePickerApp />}
+    {config.mode === 'canvas-auth-error-dialog' && (
+      <CanvasAuthErrorDialog
+        authorizeUrl={config.authorizeUrl}
+        details={config.details}
+        invalidScope={config.invalidScope}
+        scopes={config.scopes}
+      />
+    )}
   </Config.Provider>,
   rootEl
```

Comment:
We currently bundle all of the frontend apps into one bundle. This currently works OK as the bundle isn't that large. Also, when the user sees the OAuth error dialog, their browser will have already recently fetched and cached the bundle as part of the LTI launch. If however we added a bunch more mini-apps, especially ones with a substantial amount of code, we would want to split the bundle up by route.